### ISSUE TITLE
application terminates on publish after several attempts to re-open the channel

### ISSUE THREAD

**Issue by billglover** [billglover/go-adsb-console#17]
```plain
pi@raspberrypi:~ $ sudo systemctl status go-adsb-console
● go-adsb-console.service - Go ADSB Console
   Loaded: loaded (/etc/systemd/system/go-adsb-console.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2019-10-29 14:51:18 GMT; 18h ago
  Process: 4417 ExecStart=/usr/local/bin/go-adsb-console (code=exited, status=2)
 Main PID: 4417 (code=exited, status=2)

Oct 29 14:51:18 raspberrypi go-adsb-console[4417]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x37c7d0]
Oct 29 14:51:18 raspberrypi go-adsb-console[4417]: goroutine 40 [running]:
Oct 29 14:51:18 raspberrypi go-adsb-console[4417]: github.com/streadway/amqp.(*Channel).Publish(0x0, 0x1c7e500, 0x11, 0x0, 0x0, 0x0, 0x0, 0x4476b9, 0x10, 0x0, ...)
Oct 29 14:51:18 raspberrypi go-adsb-console[4417]:         /Users/bglover/go/pkg/mod/github.com/streadway/amqp@v0.0.0-20190827072141-edfb9018d271/channel.go:1331 +0x48
Oct 29 14:51:18 raspberrypi go-adsb-console[4417]: main.startUpdater.func2(0x1d7e000, 0x1d22120, 0x1d2a240, 0x4e4688, 0x1c6d420, 0x1d22000, 0x1c7e500, 0x11)
Oct 29 14:51:18 raspberrypi go-adsb-console[4417]:         /Users/bglover/dev/go-adsb-console/updater.go:83 +0x244
Oct 29 14:51:18 raspberrypi go-adsb-console[4417]: created by main.startUpdater
Oct 29 14:51:18 raspberrypi go-adsb-console[4417]:         /Users/bglover/dev/go-adsb-console/updater.go:53 +0x274
Oct 29 14:51:18 raspberrypi systemd[1]: go-adsb-console.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Oct 29 14:51:18 raspberrypi systemd[1]: go-adsb-console.service: Failed with result 'exit-code'.
```

**Comment by billglover** [comment]
This looks to be triggered by the re-connection code added as part of the fix for #16.

**Comment by billglover** [comment]
Issue not reproducible in the current version of the code.
